x-pack/solutions/security/plugins/security_solution/public/onboarding/components/onboarding_body/cards/assistant/assistant_card.tsx

```diff
@@ -39,9 +52,99 @@ export const AssistantCard: OnboardingCardComponent<AssistantCardMetadata> = ({
     setExpandedCardId(OnboardingCardId.integrations, { scroll: true });
   }, [setExpandedCardId]);
 
-  const connectors = checkCompleteMetadata?.connectors;
-  const canExecuteConnectors = checkCompleteMetadata?.canExecuteConnectors;
-  const canCreateConnectors = checkCompleteMetadata?.canCreateConnectors;
+  const [storedAssistantConnectorId, setStoredAssistantConnectorId] =
+    useStoredAssistantConnectorId(spaceId);
+
+  const [selectedConnectorId, setSelectedConnectorId] = useState(storedAssistantConnectorId || '');
+
+  const onSelectConnectorId = useCallback(
+    (connectorId: string) => {
+      setSelectedConnectorId(connectorId);
+      setStoredAssistantConnectorId(connectorId);
+    },
+    [setStoredAssistantConnectorId]
+  );
```

Comment:
Is it necessary to maintain 2 states to keep track of the connector ID? Consider using only the stored one:
```suggestion
  const [selectedConnectorId, setSelectedConnectorId] =  useStoredAssistantConnectorId(spaceId);
```
And use `setSelectedConnectorId` directly inside the `onConnectorSelected` callback.